The critical section lock _can_ be held in that place.

---
 thread-cancel.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/hurd/thread-cancel.c b/hurd/thread-cancel.c
index c8b6cf5..f463f92 100644
--- a/hurd/thread-cancel.c
+++ b/hurd/thread-cancel.c
@@ -52,7 +52,6 @@ hurd_thread_cancel (thread_t thread)
       return 0;
     }
 
-  assert (! __spin_lock_locked (&ss->critical_section_lock));
   __spin_lock (&ss->critical_section_lock);
   __spin_lock (&ss->lock);
   err = __thread_suspend (thread);
